core/src/main/scala/org/apache/spark/storage/StorageLevel.scala

```diff
@@ -131,8 +129,10 @@ class StorageLevel private(
   def description: String = {
     var result = ""
     result += (if (useDisk) "Disk " else "")
-    result += (if (useMemory) "Memory " else "")
-    result += (if (useOffHeap) "ExternalBlockStore " else "")
+    if (useMemory) {
+      if (useOffHeap) result += "Memory (off heap) "
+      else result += "Memory (heap) "
```

Comment:
probably OK to not say "Memory (heap)" since it doesn't make a ton of sense without the context of off-heap memory